WellyKorgis/project-blind-api
# Auto populdate dummy data for company

# :page_with_curl: What is this PR?
- Create application-dev.yml to separate dev environment. Init data will be loaded only when it's dev environment
- Read company data from company-data.json and store data in db
- Add relationship between Company and Company Category

# :dart: Associated ticket
#35 

# :computer: How to test
- Run `.\gradlew clean build -Pprofile=dev`
- Run `docker-compose build`
- Run `docker-compose up -d`
- To see stored data
```
docker exec -it blind-db /bin/bash

psql -U postgres -W

\c blind-db
\dt
\d company
select * from company
select * from company_category
```

# 📷 Screenshots
![image](https://user-images.githubusercontent.com/20067736/173219506-d7ee471d-c026-4324-9df2-3c55f0d8240a.png)
![image](https://user-images.githubusercontent.com/20067736/173219512-b96cb7a2-4d60-436e-a86b-72f548d4dce7.png)

# ✏️ Note
- 임의로 db에 데이터가 하나라도 있는 경우 이 기능이 실행되지 않도록 하는 로직을 넣었습니다.
- dev 환경일 때만 auto populate data 가 동작하도록 github action을 develop 브랜치에서만 동작하도록 만들면 좋겠습니다. 그러려면 develop 브랜치를 다시 가져와야 하나 싶습니다. 아니면 같은 master 이되 dev 환경 파이프라인을 따로 두거나...
